packages/wrangler/src/create-worker-preview.ts

```diff
@@ -109,6 +109,14 @@ function randomId(): string {
 	});
 }
 
+// URLs are often relative to the zone. Sometimes the base zone
+// will be grey-clouded, and so the host must be swapped out for
+// the worker route host, which is more likely to be orange-clouded
+function switchHost(originalUrl: string, host?: string): URL {
+	const url = new URL(originalUrl);
+	url.hostname = host ?? url.hostname;
```

Comment:
If host can be falsey `""` this could cause unintended behaviors

Reply:
Is there any reason not to always just use the probably-orange-clouded host?

Reply:
Sometimes we might not be able to parse an orange-clouded host (i.e. when the only route is `*/*`)

Reply:
So is there a circumstance when I'm using a wildcard route e.g. `*/*` but my root domain is grey-clouded, where this won't work?

Reply:
Yes, but there's kinda nothing we can do there if you've provided no other information. I think there's a dev.host config parameter you can add to explicitly set this, but I'll double check